app/component/favourites/favourites-panel.cjsx

```diff
@@ -0,0 +1,65 @@
+React                           = require 'react'
+Relay                           = require 'react-relay'
+NoFavouritesPanel               = require './no-favourites-panel'
+FavouriteStopCardListContainer  = require './favourite-stop-card-list-container'
+FavouriteRouteListContainer     = require './favourite-route-list-container'
+queries                         = require '../../queries'
+
+class FavouritesPanel extends React.Component
+
+  @contextTypes:
+    getStore: React.PropTypes.func.isRequired
+    executeAction: React.PropTypes.func.isRequired
+
+  getFavouriteStopContainer: (ids) =>
+    <Relay.RootContainer
+      key="fav-tops"
+      Component={FavouriteStopCardListContainer}
+      route={new queries.FavouriteStopListContainerRoute({
+        ids: ids
+        })}
+      renderLoading={-> <div className="spinner-loader"/>}
+      }
+    />
+
+  getFavouriteRouteListContainer: (ids) =>
+    <Relay.RootContainer
+      key="fav-routes"
+      Component={FavouriteRouteListContainer}
+      route={new queries.FavouriteRouteRowRoute({
+        ids: ids
+        })}
+      renderLoading={-> <div className="spinner-loader"/>}
+      }
+    />
+
+  componentDidMount: ->
+    @context.getStore('FavouriteRoutesStore').addChangeListener @onChange
+
+  componentWillUnmount: ->
+    @context.getStore('FavouriteRoutesStore').removeChangeListener @onChange
+
+  onChange: (id) =>
+    @forceUpdate()
+
+  getFavourites: (stops, routes) =>
```

Comment:
Is this really needed, couldn't we just have 

  render: ->
    FavouriteStopsStore = @context.getStore 'FavouriteStopsStore'
    FavouriteRoutesStore = @context.getStore 'FavouriteRoutesStore'

    if stops.concat(routes).length==0
      <NoFavouritesPanel/>
    else
      <div>
        {@getFavouriteStopContainer FavouriteStopsStore.getStops()}
        {@getFavouriteRouteListContainer FavouriteRoutesStore.getRoutes()}
      </div>